cmake/tribits/core/package_arch/TribitsAddTestHelpers.cmake

```diff
@@ -617,6 +617,32 @@ FUNCTION(TRIBITS_SET_DISABLED_AND_MSG  TEST_NAME_IN  PARSE_DISABLED
 ENDFUNCTION()
 
 
+#
+# Read ${TEST_NAME_IN}_SET_RUN_SERIAL and set output
+# var SET_RUN_SERIAL_OUT
+#
+# Usage:
+#
+#    TRIBITS_SET_RUN_SERIAL(
+#      <testName>              # The full test name passed to add_test()
+#      "${PARSE_RUN_SERIAL}"   # From the input option "RUN_SERIAL"
+#      <setRunSerial>          # Sets var of this name
+#      )
+#
+FUNCTION(TRIBITS_SET_RUN_SERIAL  TEST_NAME_IN  PARSE_RUN_SERIAL
+  SET_RUN_SERIAL_OUT
+  )
+
+  SET(SET_RUN_SERIAL "${PARSE_RUN_SERIAL}")
+  IF (NOT "${${TEST_NAME_IN}_SET_RUN_SERIAL}" STREQUAL "")
```

Comment:
Yes, you want `${TEST_NAME_IN}_SET_RUN_SERIAL` to override the input `RUN_SERIAL`, true or false.  Look at the unit test cases.  If you change this to:

```
IF (${TEST_NAME_IN}_SET_RUN_SERIAL)
```

you will see that one of the unit tests fails.

Reply:
This may be a source of confusion from a user/usability perspective. Is this override behavior documented anywhere?

Reply:
I'm ready to merge once this behavior is documented and a warning is generated upon override.

Reply:
Yes, see updated documentation starting at:

* https://github.com/trilinos/Trilinos/pull/8063/files#diff-09fda8e3c6d440bb6132c674d5fe1cbfR2061

Reply:
> I'm ready to merge once this behavior is documented both in the source and in the user-facing documentation and a warning is generated upon override.

The idiom:

```
IF ("${<var_name>}" STREQUAL "")
```

is very common and it means "if `<var_name> is set to any value".  You can see this looking at:

```
$ cd TriBITS/tribits/core/package_arch/

$ grep -nHi 'STREQUAL ""' * | wc -l
120
```

So should we add comments to 120 if blocks?  A better solution might be to add a section (perhaps in the appendix) to the TriBITS Developers Guide with common idioms used in the implementation of TriBITS?

No warning is needed because this is the documented and expected behavior as documented in:

* https://github.com/trilinos/Trilinos/pull/8063/files/90cee2a4bce5fc408cdb008aefa71df24dc18198#diff-09fda8e3c6d440bb6132c674d5fe1cbfR2043

and in the unit tests:

* https://github.com/bartlettroscoe/TriBITS/blob/e8467f7117b7e1b33ba9dfcd3c4750e1330d3acb/test/core/TestingFunctionMacro_UnitTests.cmake#L1240
* https://github.com/bartlettroscoe/TriBITS/blob/e8467f7117b7e1b33ba9dfcd3c4750e1330d3acb/test/core/TestingFunctionMacro_UnitTests.cmake#L3804

And what is interesting about this is that even without looking at the documentation and the unit tests, you still were able to interpret its meaning as per you [above statement](https://github.com/trilinos/Trilinos/pull/8063#discussion_r493609955):

> This predicate is true even when `${TEST_NAME_IN}_SET_RUN_SERIAL=OFF`. When this predicate is true, it will override the `RUN_SERIAL` options to the tribits add test cmake function

Exactly!

We don't need a ton of extra "warnings" printed to the screen for this.  That would be very bad to do.  (TriBITS puts out enough "notes" already and stuff like this would just be spam.)

@jmgate, what is your opinion on this after looking at the documentation:

* https://tribits.org/doc/TribitsBuildReference.html#set-specific-tests-to-run-in-serial

?

Reply:
@e10harvey and @jmgate, let's continue any review of the RUN_SERIAL code to the new PR https://github.com/TriBITSPub/TriBITS/pull/332, but I will argue that this should not hold up the merge of this Trilinos PR because any suggested changes will only impact internal documentation and not external documentation or behavior.